crates/settlement-clients/ethereum/src/conversion.rs

```diff
@@ -1,12 +1,325 @@
+    data.into_iter().fold(String::new(), |mut output, arr| {
+        // Convert the array to a hex string
+        let hex = arr.iter().fold(String::new(), |mut output, byte| {
+            let _ = write!(output, "{byte:02x}");
+            output
+        });
+
+        // Ensure the hex string is exactly 64 characters (32 bytes)
+        let _ = write!(output, "{hex:0>64}");
+        output
+    })
+}
+
+pub(crate) fn u8_48_to_hex_string(data: [u8; 48]) -> String {
+    // Split the array into two parts
+    let (first_32, last_16) = data.split_at(32);
+
+    // Convert and pad each part
+    let first_hex = to_padded_hex(first_32);
+    let second_hex = to_padded_hex(last_16);
+
+    // Concatenate the two hex strings
+    first_hex + &second_hex
+}
+
+/// To prepare the sidecar for EIP 4844 transaction
+pub(crate) async fn prepare_sidecar(
+    state_diff: &[Vec<u8>],
+    trusted_setup: &KzgSettings,
+) -> EyreResult<(Vec<FixedBytes<BYTES_PER_BLOB>>, Vec<FixedBytes<48>>, Vec<FixedBytes<48>>)> {
+    let mut sidecar_blobs = vec![];
+    let mut sidecar_commitments = vec![];
+    let mut sidecar_proofs = vec![];
+
+    for blob_data in state_diff {
+        let fixed_size_blob: [u8; BYTES_PER_BLOB] = blob_data.as_slice().try_into()?;
+
+        let blob = Blob::new(fixed_size_blob);
+
+        let commitment = KzgCommitment::blob_to_kzg_commitment(&blob, trusted_setup)?;
+        let proof = KzgProof::compute_blob_kzg_proof(&blob, &commitment.to_bytes(), trusted_setup)?;
+
+        sidecar_blobs.push(FixedBytes::new(fixed_size_blob));
+        sidecar_commitments.push(FixedBytes::new(commitment.to_bytes().into_inner()));
+        sidecar_proofs.push(FixedBytes::new(proof.to_bytes().into_inner()));
+    }
+
+    Ok((sidecar_blobs, sidecar_commitments, sidecar_proofs))
+}
+
+#[cfg(test)]
+mod tests {
+
+    use super::*;
+    use color_eyre::eyre::eyre;
+    use rstest::rstest;
+    use std::{fs, path::Path};
+
+    #[rstest]
+    #[case::typical(&[
+        0x00, 0x11, 0x22, 0x33, 0x44, 0x55, 0x66, 0x77,
+        0x88, 0x99, 0xAA, 0xBB, 0xCC, 0xDD, 0xEE, 0xFF,
+        0x00, 0x11, 0x22, 0x33, 0x44, 0x55, 0x66, 0x77,
+        0x88, 0x99, 0xAA, 0xBB, 0xCC, 0xDD, 0xEE, 0xFF
+    ], U256::from_str_radix("00112233445566778899AABBCCDDEEFF00112233445566778899AABBCCDDEEFF", 16).unwrap())]
+    #[case::minimum(&[0; 32], U256::ZERO)]
+    #[case::maximum(&[0xFF; 32], U256::MAX)]
+    #[case::short(&[0xFF; 16], U256::from_be_slice(&[0xFF; 16]))]
+    #[case::empty(&[], U256::ZERO)]
+    fn slice_u8_to_u256_works(#[case] slice: &[u8], #[case] expected: U256) {
+        match slice_u8_to_u256(slice) {
+            Ok(response) => {
+                assert_eq!(response, expected);
+            }
+            Err(e) => {
+                panic!("{}", e);
+            }
+        }
+    }
+
+    #[rstest]
+    #[case::over(&[0xFF; 33])]
+    fn slice_u8_to_u256_panics(#[case] slice: &[u8]) {
```

Comment:
I think you can add #[should_panic] on top of a specific #[case] and it only applies to that, we can combine this and the above one?

Reply:
We can, but how do you think the naming of the function be, right now there is a clear decoupling between failing and passing cases naming separating two types of tests.

WDYT ?

Reply:
hmm, I guess we can do something like `slice_u8_to_u256_all_cases` or `slice_u8_to_u256_working_and_failing_cases`

Reply:
Alright, Implemented.